hearpreprocess/pipeline.py

```diff
@@ -754,10 +767,16 @@ def run(self):
             max_task_duration_by_split = self.task_config["max_task_duration_by_split"]
             # Check if all the splits are specified in
             # max_task_duration_by_split
-            assert set(max_task_duration_by_split.keys()) == set(SPLITS)
+            assert set(max_task_duration_by_split.keys()) == set(
+                self.task_config["splits"]
+            )
             max_split_duration = max_task_duration_by_split[self.split]
         else:
             # Get the default values for the split
+            assert "folds" not in self.task_config, "Default max split duration "
+            "cannot be used for tasks with folds. Please define max_split_duration "
+            "explicitly for each fold. Set each fold's max_split_duration to None if "
+            "the full fold is required"
```

Comment:
I'm thinking about what task_config is going to look like now -- so we need to add in a list of "folds" ex: `'folds': ['fold1', 'fold2', ...]` - this is essentially what defines a task as being k-fold. But we also need to add `max_task_duration_by_split` for each fold, keyed on the same names, i.e `['fold1', 'fold2', ...]` - correct?

Reply:
Maybe this is fine. But I'm curious whether this can be simplified at all. Maybe we add back the idea of "split-mode" that can take on one of two values: `"kfold"` or `"predefined`". Or like just a boolean that is `"kfold": True/False`. Perhaps we also add the "splits" key directly to the task_metadata. For non-kfold then we assert it is train/val/test?